src/rag.py

```
@@ -0,0 +1,98 @@
+    )
+    faiss_retriever = vectorstore.as_retriever(
+        search_type="mmr",
+        search_kwargs={"k": TOP_K, "fetch_k": FETCH_K},
+    )
+
+    bm25_retriever = None
+    if docs_cache_path and os.path.exists(docs_cache_path):
+        # docs_cache.pkl は ingest.py がローカルの Obsidian Vault から生成する
+        # 信頼済みローカルファイルのため pickle デシリアライズは安全
+        with open(docs_cache_path, "rb") as f:
+            all_docs = pickle.load(f)
```

Ruff (0.15.7)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)